bayes_opt/constraint.py

```diff
@@ -167,6 +182,12 @@ def predict(self, X):
         """
         X_shape = X.shape
         X = X.reshape((-1, self._model[0].n_features_in_))
+
+        result: NDArray[Float]
+        y_mean: NDArray[Float]
+        y_std: NDArray[Float]
+        p_lower: NDArray[Float]
+        p_upper: NDArray[Float]
```

Comment:
Why not declare the types when assigning?

Reply:
If a variable is assigned in the following ways(unpacking tuple),
the type declaration cannot be made along with the assignment

https://github.com/bayesian-optimization/BayesianOptimization/blob/2298f7c5c9c7b6c9b64669bacd0e87f9ba911f96/bayes_opt/constraint.py#L170-L172